R/schedule.R

```diff
@@ -21,132 +21,109 @@ get_tune_schedule <- function(wflow, param, grid) {
 		cli::cli_abort("Argument {.arg grid} must be a tibble.")
 	}
 
-	# ----------------------------------------------------------------------------
-	# Get information on the parameters associated with the supervised model
+	# Which parameter belongs to which stage and which is a submodel parameter?
+	param_info <- get_param_info(wflow)
 
-	model_spec <- extract_spec_parsnip(wflow)
-	model_type <- class(model_spec)[1]
-	model_eng <- model_spec$engine
-
-	# Which, if any, is a submodel
-	model_param <- parsnip::get_from_env(paste0(model_type, "_args")) %>%
-		dplyr::filter(engine == model_spec$engine) %>%
-		dplyr::select(name = parsnip, has_submodel)
-
-	# Merge the info in with the other parameters
-	param <- dplyr::left_join(param, model_param, by = "name") %>%
-		dplyr::mutate(
-			has_submodel = dplyr::if_else(is.na(has_submodel), FALSE, has_submodel)
-		)
-
-	# ------------------------------------------------------------------------------
-	# Get tuning parameter IDs for each stage of the workflow
-
-	if (any(param$source == "recipe")) {
-		pre_id <- param$id[param$source == "recipe"]
-	} else {
-		pre_id <- character(0)
-	}
+	schedule <- schedule_stages(grid, param_info, wflow)
 
-	if (any(param$source == "model_spec")) {
-		model_id <- param$id[param$source == "model_spec"]
-		sub_id <- param$id[param$source == "model_spec" & param$has_submodel]
-		non_sub_id <- param$id[param$source == "model_spec" & !param$has_submodel]
-	} else {
-		model_id <- sub_id <- non_sub_id <- character(0)
-	}
-
-	if (any(param$source == "tailor")) {
-		post_id <- param$id[param$source == "tailor"]
+	og_cls <- class(schedule)
+	if (nrow(param) == 0) {
```

Comment:
It's a little gross to code here but this class structure feels like a good solution (until we know that it is not)